src/script/components/LexicalInput/LexicalInput.tsx

```diff
@@ -0,0 +1,182 @@
+  saveDraftState: (editor: string) => void;
+  loadDraftState: () => Promise<DraftState>;
+  mentionCandidates: User[];
+  onShiftTab: () => void;
+}
+
+export const LexicalInput = forwardRef<LexicalEditor, LexicalInputProps>(
+  (
+    {
+      placeholder,
+      propertiesRepository,
+      searchRepository,
+      inputValue,
+      setInputValue,
+      children,
+      hasLocalEphemeralTimer,
+      saveDraftState,
+      loadDraftState,
+      editMessage,
+      mentionCandidates,
+      onShiftTab,
+    },
+    ref,
+  ) => {
+    // Emojis
+    const [shouldReplaceEmoji, setShouldReplaceEmoji] = useState<boolean>(
+      propertiesRepository.getPreference(PROPERTIES_TYPE.EMOJI.REPLACE_INLINE),
+    );
+
+    const editorConfig: InitialConfigType = {
+      namespace: 'WireLexicalEditor',
+      theme,
+      onError(error: unknown) {
+        logger.error(error);
+        throw error;
+      },
+      nodes: [BeautifulMentionNode],
+    };
+
+    const queryMentions = (queryString?: string | null) => {
+      return queryString ? searchRepository.searchUserInSet(queryString, mentionCandidates) : mentionCandidates;
+    };
+
+    const onChange = useCallback(
+      (editorState: EditorState, lexicalEditor: LexicalEditor) => {
+        lexicalEditor.registerTextContentListener(textContent => {
+          setInputValue(textContent);
+        });
+
+        const stringifyEditor = JSON.stringify(editorState.toJSON());
+        saveDraftState(stringifyEditor);
+      },
+      [saveDraftState, setInputValue],
+    );
+
+    useEffect(() => {
+      amplify.subscribe(WebAppEvents.PROPERTIES.UPDATE.EMOJI.REPLACE_INLINE, setShouldReplaceEmoji);
+      amplify.subscribe(WebAppEvents.PROPERTIES.UPDATED, (properties: WebappProperties) => {
+        setShouldReplaceEmoji(properties.settings.emoji.replace_inline);
+      });
+    }, []);
+
+    return (
+      <LexicalComposer initialConfig={editorConfig}>
+        <div className="controls-center">
+          <div className={cx('input-bar--wrapper')}>
+            <AutoFocusPlugin />
+            <GlobalEventsPlugin onShiftTab={onShiftTab} />
+            <EditorRefPlugin editorRef={ref} />
+            <DraftStatePlugin setInputValue={setInputValue} loadDraftState={loadDraftState} />
+            <EditMessagePlugin onMessageEdit={editMessage} />
+
+            {shouldReplaceEmoji && <EmojiPickerPlugin />}
+
+            <PlainTextPlugin
+              contentEditable={
+                <ContentEditable
+                  value={inputValue}
+                  className="conversation-input-bar-text"
+                  data-uie-name="input-message"
+                />
+              }
+              placeholder={<Placeholder text={placeholder} hasLocalEphemeralTimer={hasLocalEphemeralTimer} />}
+              ErrorBoundary={LexicalErrorBoundary}
+            />
+
+            <BeautifulMentionsPlugin onSearch={queryMentions} />
+
+            <OnChangePlugin onChange={onChange} />
```

Comment:
can you give the onChange function a meaningful name which describes the action we are doing onchange?

Reply:
Fixed :)